packages/cli/src/util/token.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { consola } from "consola";
+import * as fs from "node:fs";
+import path from "node:path";
+
+const TOKEN_ENV_VARS = ["FOUNDRY_TOKEN", "FOUNDRY_SDK_AUTH_TOKEN"] as const;
+
+/**
+ * Loads a JWT Auth Token from an argument, a file, or environment variable returning the first found.
+ * @param token The token as a string.
+ * @param tokenFile The path to the token file.
+ * @returns The token as a string.
+ * @throws An error if no token is found.
+ */
+export function loadToken(token?: string, tokenFile?: string): string {
```

Comment:
Should we make all of these async rather than sync for consistency with config loading?